config.c

```diff
@@ -188,22 +189,17 @@ void read_FTLconf(void)
 
 	// BLOCKINGMODE
 	// defaults to: MODE_IP
-	config.blockingmode = MODE_IP;
-	buffer = parse_FTLconf(fp, "BLOCKINGMODE");
-
-	if(buffer != NULL)
-	{
-		if(strcasecmp(buffer, "NXDOMAIN") == 0)
-			config.blockingmode = MODE_NX;
-	}
-
+	get_blocking_mode(fp);
 	switch(config.blockingmode)
 	{
 		case MODE_NX:
 			logg("   BLOCKINGMODE: NXDOMAIN for blocked domains");
 			break;
+		case MODE_NULL:
+			logg("   BLOCKINGMODE: Null IPs for blocked domains");
+			break;
 		default:
-			logg("   BLOCKINGMODE: Pi-hole's IP for blocked domains");
+			logg("   BLOCKINGMODE: Pi-hole's IPs for blocked domains");
 			break;
```

Comment:
You should probably use a specific switch statement for `MODE_IP` and keep a default for errors.

Reply:
There can be no error. The three possible cases are:
`BLOCKINGMODE=NXDOMAIN`, or `BLOCKINGMODE=NULL` or anything else (even missing option). In the latter case, it will always fall back to the legacy behavior (IP address of Pi-hole). There is no situation that we can consider as failed.